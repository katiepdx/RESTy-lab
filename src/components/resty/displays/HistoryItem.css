.historyItem {
  background-color: rgb(134, 65, 65);
  font-size: 16px;
  padding: 0 12px;
  width: 100%;
  color: aliceblue;
  border-radius: 0.8rem;
}

.historyItem:hover {
  background-color: rgba(134, 65, 65, 0.795);
}
