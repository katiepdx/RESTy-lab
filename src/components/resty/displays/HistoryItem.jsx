import React from 'react';
import propTypes from 'prop-types';

const HistoryItem = ({ url, method }) => (
  <section>
    {/* search history */}
    <div>
      <p>{method}</p>
      <p>{url}</p>
    </div>
  </section>
);

HistoryItem.propType = {
  url: propTypes.string.isRequired,
  method: propTypes.string.isRequired
};

export default HistoryItem;
