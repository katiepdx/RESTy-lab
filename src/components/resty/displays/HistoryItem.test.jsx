import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryItem from './HistoryItem'

describe('HistoryItem component', () => {
  afterAll(() => cleanup());

  it('renders HistoryItem', () => {
    const { asFragment } = render(<HistoryItem
      url="www.url.com"
      method="GET"
    />);
    expect(asFragment()).toMatchSnapshot();
  })
});
