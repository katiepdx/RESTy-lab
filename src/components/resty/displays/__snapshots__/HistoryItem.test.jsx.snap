// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component renders HistoryItem 1`] = `
<DocumentFragment>
  <section
    class="historyItem"
  >
    <p>
      Method: GET
    </p>
    <p>
      URL: www.url.com
    </p>
  </section>
</DocumentFragment>
`;
